### ISSUE TITLE
Sceptre cannot resolve stack_output

### ISSUE THREAD

**Issue by lukeplausin** [Sceptre/sceptre#716]
Using sceptre v2 on windows (git bash shell).

I believe this issue may be related to my OS or shell. 

I have two templates set up where one template has a dependency to the other template. Template A defines an output `myOutput`, which is referenced in parameters of template B using the `stack_output` resolver.

File tree:
```
├───config
│   └───default
│       ├───template_a.yaml
│       └───myapp
│            └───template_b.yaml
├───scripts
└───templates
         ├───template_a.yaml
         └───template_b.yaml
```

Template B config:
```yaml
template_path: template_b.yaml
parameters:
  VpcId: !stack_output default/template_a.yaml::myOutput
```

The config in template B gives the path to the config in template A, including the directory name, file extension and namespace separator :: as per the docs.

Stack A launches without any issues, but when I launch stack B with this command, sceptre throws a python exception:

```
$ sceptre --debug launch default/cdc/compute
[2019-05-29 11:30:42] - Adding yaml constructors for the entry point groups ['sceptre.hooks', 'sceptre.resolvers']
[2019-05-29 11:30:42] - Added constructor for <class 'sceptre.hooks.asg_scaling_processes.ASGScalingProcesses'> with node tag !asg_scheduled_actions

< OUTPUT OMITTED >

[2019-05-29 11:30:46] - default/myapp/template_b - Launching Stack
[2019-05-29 11:30:46] - default/myapp/template_b - Stack is in the ?[36mPENDING?[0m state
[2019-05-29 11:30:46] - default/myapp/template_b - Creating Stack
[2019-05-29 11:30:46] - Resolving Stack output: default/template_a.yaml::myOutput
Traceback (most recent call last):
  File "c:\bin\python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\bin\python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\bin\python37\Scripts\sceptre.exe\__main__.py", line 9, in <module>
  File "c:\bin\python37\lib\site-packages\click\core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "c:\bin\python37\lib\site-packages\click\core.py", line 717, in main
    rv = self.invoke(ctx)
  File "c:\bin\python37\lib\site-packages\click\core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\bin\python37\lib\site-packages\click\core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\bin\python37\lib\site-packages\click\core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "c:\bin\python37\lib\site-packages\click\decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "c:\bin\python37\lib\site-packages\sceptre\cli\helpers.py", line 35, in decorated
    return func(*args, **kwargs)
  File "c:\bin\python37\lib\site-packages\sceptre\cli\launch.py", line 38, in launch_command
    responses = plan.launch()
  File "c:\bin\python37\lib\site-packages\sceptre\plan\plan.py", line 130, in launch
    return self._execute(*args)
  File "c:\bin\python37\lib\site-packages\sceptre\plan\plan.py", line 40, in _execute
    return executor.execute(*args)
  File "c:\bin\python37\lib\site-packages\sceptre\plan\executor.py", line 55, in execute
    stack, status = future.result()
  File "c:\bin\python37\lib\concurrent\futures\_base.py", line 425, in result
    return self.__get_result()
  File "c:\bin\python37\lib\concurrent\futures\_base.py", line 384, in __get_result
    raise self._exception
  File "c:\bin\python37\lib\concurrent\futures\thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "c:\bin\python37\lib\site-packages\sceptre\plan\executor.py", line 62, in _execute
    result = getattr(actions, self.command)(*args)
  File "c:\bin\python37\lib\site-packages\sceptre\plan\actions.py", line 185, in launch
    status = self.create()
  File "c:\bin\python37\lib\site-packages\sceptre\hooks\__init__.py", line 104, in decorated
    response = func(self, *args, **kwargs)
  File "c:\bin\python37\lib\site-packages\sceptre\plan\actions.py", line 61, in create
    "Parameters": self._format_parameters(self.stack.parameters),
  File "c:\bin\python37\lib\site-packages\sceptre\resolvers\__init__.py", line 75, in __get__
    resolve, getattr(instance, self.name), Resolver
  File "c:\bin\python37\lib\site-packages\sceptre\helpers.py", line 63, in _call_func_on_values
    func_on_instance(key)
  File "c:\bin\python37\lib\site-packages\sceptre\helpers.py", line 57, in func_on_instance
    func(attr, key, value)
  File "c:\bin\python37\lib\site-packages\sceptre\resolvers\__init__.py", line 71, in resolve
    attr[key] = value.resolve()
  File "c:\bin\python37\lib\site-packages\sceptre\resolvers\stack_output.py", line 128, in resolve
    stack for stack in self.stack.dependencies if stack.name == friendly_stack_name
StopIteration
```

Could this be another issue related to forward slashes becoming backslashes on Windows?

Thanks

Luke

**Comment by ngfgrant** [comment]
Hey @lukeplaysin there was some issues with python 3.7 during testing so it is not officially supported just yet. It might be part of the issue? 

Might also be something in the asg resolver. On the face of it it doesn’t look like a windows thing.

**Comment by lukeplausin** [comment]
Ah ok! Maybe I'm a bit keen going straight for 3.7 then. I'll try to work out what's breaking in 3.7

**Comment by ngfgrant** [comment]
There is a couple of PRs for 3.7 support atm. If you try run it with the 3.6 it might work?

**Comment by lukeplausin** [comment]
The problem also persists on version 3.6.8

**Comment by ngfgrant** [comment]
Are you able to share the project at all?

**Comment by lukeplausin** [comment]
Sorry I can't share project code

I've done a little investigation today. I put some print statements into the stack_output resovler, it looks like the problem is indeed caused by backslashes in windows.

```
[2019-06-03 10:01:54] - default/vpc - Successfully initiated creation of Change Set 'newdomain'
Have dependency_stack_name == default\vpc.yaml
Looking for friendly_stack_name == default\vpc
found stack.name == default/vpc
Traceback (most recent call last):
  File "C:\Users\luke.plausin\.virtualenvs\sceptre\Scripts\sceptre-script.py", line 11, in <module>
............
  File "c:\users\luke.plausin\code\sceptre\sceptre\resolvers\stack_output.py", line 133, in resolve
    stack for stack in self.stack.dependencies if stack.name == friendly_stack_name
StopIteration
```

The dependency stack name is set on line 112
`self.dependency_stack_name = normalise_path(dep_stack_name)`

Is `normalise_path` supposed to change backslashes to forward slashes as `sceptreise_path` does? Would using `sceptreise_path` as well be a suitable fix?

**Comment by vadym** [comment]
I've catch up the same error.

Python 2.7.15

**Comment by ngfgrant** [comment]
Hey thanks to Luke it will be fixed 2.1.4

**Comment by zaro0508** [comment]
scepter is now at ver 2.7.x.  Should have been fixed somewhere along the way.  please re-open if it's still a problem.
